hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -821,7 +893,7 @@ private ByteString readStateMachineData(
   public CompletableFuture<Void> flush(long index) {
     return CompletableFuture.allOf(
         writeChunkFutureMap.entrySet().stream().filter(x -> x.getKey() <= index)
-            .map(Map.Entry::getValue).toArray(CompletableFuture[]::new));
+            .map(e -> e.getValue().getRaftFuture()).toArray(CompletableFuture[]::new));
```

Comment:
Since it is a `NavigableMap`, we should use `headMap`.
```java
  public CompletableFuture<Void> flush(long index) {
    final SortedMap<Long, WriteFutureContext> head = writeChunkFutureMap.headMap(index + 1);
    if (head.isEmpty()) {
      return CompletableFuture.completedFuture(null);
    }
    return CompletableFuture.allOf(head.values().stream()
        .map(WriteFutureContext::getRaftFuture)
        .toArray(CompletableFuture[]::new));
  }
```